### PR TITLE
change docker build to alpine

### PR DESCRIPTION
**Pull request by mmacata**
This PR changes the docker base-image from Ubuntu to alpine, cleans up requirements and uses multistage build.
Docker image size is reduced from `1.13GB` to `246MB` and software components are updated:

| Before (Ubuntu) | With this PR (alpine) |
| - | - | 
| Python 3.6.9 | Python 3.8.7 | 
| GDAL 2.2.3 |  GDAL 3.1.4 | 
| PROJ 4.9.3 | PROJ 7.1.1 | 

Also this PR copies the code from build context instead of using `git clone` and uses the version from git tags instead of setting it manually with the `SETUPTOOLS_SCM_PRETEND_VERSION` environment variable.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -75,7 +75,7 @@ An active internet connection is required. All requests to the openEO GRASS GIS
 
         .. code-block:: bash
 
-            docker build -t openeo_grass_gis_driver docker
+            docker build -f docker/Dockerfile -t openeo_grass_gis_driver .
```
Comment (metzm) [inline docs/index.rst:78]:
```suggestion
            docker build -f docker/Dockerfile -t openeo_grass_gis_driver .
```

File: src/openeo_grass_gis_driver/main.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 __email__ = "soerengebbert@googlemail.com"
 
 
+create_endpoints()
+
 if __name__ == '__main__':
-    create_endpoints()
```
Comment (metzm) [inline src/openeo_grass_gis_driver/main.py:14]: Why are you moving `create_enpoints()` out?

**Comment by metzm** [comment]
Thanks for the substantial size reduction! I have one suggestion and one question, otherwise ready to merge.

File: src/openeo_grass_gis_driver/main.py (hunk 2 above)
Comment (mmacata) [reply src/openeo_grass_gis_driver/main.py:14]: Now the more stable gunicorn wsgi server is used to run the openeo app. This way, `main.py` is not called directly anymore (before done via `python3 -m openeo_grass_gis_driver.main`) and `if __name__ == '__main__':` does not match anymore. This is the same mechanism [actinia_core](https://github.com/mundialis/actinia_core/blob/master/src/actinia_core/main.py) is using for stable production server while still be able to debug locally.

File: docker/Dockerfile [hunk 3]
```diff
@@ -1,41 +1,28 @@
...
-RUN pip3 install -r requirements.txt
-
-ENV SETUPTOOLS_SCM_PRETEND_VERSION=0.0.1
-ENV SETUPTOOLS_SCM_DEBUG=1
-RUN make install
 RUN make test
 
-CMD python3 -m openeo_grass_gis_driver.main
+FROM install as app
+ENV GDAL_CACHEMAX=2000
```
Comment (neteler) [inline docker/Dockerfile:26]: Suggestion: rather than hardcoded GB use % (?)

`export GDAL_CACHEMAX="25%"    # default 5% of RAM if GDAL_CACHEMAX unset`

Comment (mmacata) [reply docker/Dockerfile]: thx, I changed it.

**Comment by mmacata** [comment]
Thanks for review, I will merge now!

**Pull request merged**